pkg/controller/control.go

```diff
@@ -213,6 +213,14 @@ func (c *Controller) Expand(size int64) error {
 	go func(size int64) {
 		expanded := false
 		defer func() {
+			// Frontend expansion involves in the iSCSI session rescanning, which will wait for the in-fly io requests complete.
+			// Hence there will be a deadlock once we use the lock to protect this frontend expansion.
+			if c.frontend != nil {
+				if err := c.frontend.Expand(size); err != nil {
+					logrus.WithError(err).Error("Failed to expand the frontend")
+					expanded = false
+				}
```

Comment:
return here if any error?

Reply:
No. We need to rely on the below function `c.finishExpansion(expanded, size)` to update flag `c.isExpanding`

Reply:
Got it. Thank you.

Reply:
Will this one be a halfway expansion? 

Next time when volume detaches and reattaches, will the size of replicas be different than the size of the engine?

Reply:
Actually there is a separate issue for this: longhorn/longhorn#3601.
The key point is that we cannot expand the frontend if there is no replica succeeding the expansion. Otherwise, you will see tgt r/w errors, which may need to data loss/